server/graphql/v2/mutation/MemberMutations.js

```diff
@@ -0,0 +1,56 @@
+import { GraphQLNonNull, GraphQLString } from 'graphql';
+
+import models from '../../../models';
+import { NotFound, Unauthorized } from '../../errors';
+import { idDecode } from '../identifiers';
+import { AccountReferenceInput } from '../input/AccountReferenceInput';
+import { Member } from '../object/Member';
+
+const memberMutations = {
+  editPublicMessage: {
+    type: new GraphQLNonNull(Member),
+    description: 'Edit the public message for the given Member of a Collective',
+    args: {
+      FromCollectiveLegacyId: {
+        type: AccountReferenceInput,
+        description: 'Legacy id for the donating Collective',
+      },
+      ToCollectiveId: {
+        type: AccountReferenceInput,
+        description: 'Public id for the receiving Collective',
+      },
+      message: {
+        type: GraphQLString,
+        description: 'New public message',
+      },
+    },
+    async resolve(_, args, req) {
+      const { FromCollectiveLegacyId, ToCollectiveId, message } = args;
+
+      if (!req.remoteUser || !req.remoteUser.isAdmin(FromCollectiveLegacyId)) {
+        throw new Unauthorized("You don't have the permission to edit this member's public message");
+      }
+
+      const decodedToCollectiveId = idDecode(ToCollectiveId.id, 'account');
+
+      const member = await models.Member.findOne({
+        where: {
+          MemberCollectiveId: FromCollectiveLegacyId.legacyId,
+          CollectiveId: decodedToCollectiveId,
+        },
+      });
+
+      if (!member) {
+        throw new NotFound('No member found');
+      }
+
+      await member.update({
+        publicMessage: message,
+      });
+
+      return member;
```

Comment:
You can directly write

```suggestion
      return member.update({ publicMessage: message });
```